packages/core/saved-objects/core-saved-objects-migration-server-internal/src/actions/es_errors.ts

```diff
@@ -7,34 +7,42 @@
  * License v3.0 only", or the "Server Side Public License, v 1".
  */
 
-import type * as estypes from '@elastic/elasticsearch/lib/api/typesWithBodyKey';
+import type { ErrorCause } from '@elastic/elasticsearch/lib/api/types';
 
-export const isWriteBlockException = (errorCause?: estypes.ErrorCause): boolean => {
+export const isWriteBlockException = (errorCause?: ErrorCause): boolean => {
   return (
     errorCause?.type === 'cluster_block_exception' &&
-    errorCause?.reason?.match(/index \[.+] blocked by: \[FORBIDDEN\/8\/.+ \(api\)\]/) !== null
+    hasAllKeywordsInOrder(errorCause?.reason, ['index', 'blocked by: ', '[FORBIDDEN/8', '(api)'])
```

Comment:
One other approach, instead of changing the implementation could be to assert that the input length is less than e.g. `<5_000`. Would that satisfy the code scanner?